api/v1alpha1/envoygateway_types.go

```diff
@@ -582,6 +582,18 @@ type ExtensionHooks struct {
 type XDSTranslatorHooks struct {
 	Pre  []XDSTranslatorHook `json:"pre,omitempty"`
 	Post []XDSTranslatorHook `json:"post,omitempty"`
+
+	// Translation defines the configuration for the translation hook.
+	Translation *TranslationConfig `json:"translation,omitempty"`
+}
+
+// TranslationConfig defines the configuration for the translation hook.
+type TranslationConfig struct {
+	// IncludeAll defines whether all resources should be included in the translation hook.
+	// Default resources are clusters and secrets. When enabled, listeners and routes are also included.
+	//
+	// +optional
+	IncludeAll *bool `json:"includeAll,omitempty"`
```

Comment:
I do prefer @guydc 's suggestion here to break this up into `IncludeRoutes` && `IncludeListeners`, bonus points if you can add `ExcludeClusters` and `ExcludeSecrets`

Reply:
cc @guydc

Reply:
the exclude here looks weired

Reply:
I prefer include for all. If someone is configuring the "translationconfig", they are no longer subject to the previous defaults. We should validate that atleast one of them is included.  If it's nil - previous defaults.

Reply:
And, for future use cases, maybe make this a struct instead of a bool, so that we can later add selectors and such. 

like: 
```
        xdsTranslator:
          translation:
            routes: 
              all: true
            listeners: 
              labelSelector:
                foo: bar
```

We'll only implement the "all" bit for now.

Reply:
this is fine, it looks like we are not leaving room for a selector which will live in the same level and we'll need to use CEL to avoid setting both fields

Reply:
+1 to @guydc's suggestion

Reply:
all is not specific for semantic meaning, how about:

```
        xdsTranslator:
          translation:
            routes:
              includeAll: true
            listeners: 
              includeAll: true
            clusters:  
              includeAll: true
            secrets:
              includeAll: true
```
? @guydc @arkodg